examples/ollama-hugging-face/src/test/java/com/example/ollamahf/OllamaHuggingFaceContainer.java

```diff
@@ -0,0 +1,96 @@
+package com.example.ollamahf;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import org.testcontainers.containers.ContainerLaunchException;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.ollama.OllamaContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.IOException;
+
+public class OllamaHuggingFaceContainer extends OllamaContainer {
+
+    private final String imageName;
+
+    private final HuggingFaceModel huggingFaceModel;
+
+    public OllamaHuggingFaceContainer(String imageName, HuggingFaceModel model) {
+        super(DockerImageName.parse("ollama/ollama:0.1.42"));
+        this.imageName = imageName;
+        this.huggingFaceModel = model;
+    }
+
+    @Override
+    protected void configure() {
+        super.configure();
+
+        if (huggingFaceModel != null) {
+            this.setImage(
+                    new ImageFromDockerfile()
+                        .withDockerfileFromBuilder(builder -> {
+                            builder
+                                .from(this.getDockerImageName())
+                                .run("apt-get update && apt-get upgrade -y && apt-get install -y python3-pip")
+                                .run("pip install huggingface-hub")
+                                .build();
+                        })
+                );
+        }
+    }
+
+    @Override
+    protected void containerIsStarted(InspectContainerResponse containerInfo) {
+        super.containerIsStarted(containerInfo);
```

Comment:
```suggestion
        super.containerIsStarted(containerInfo, reused);
         if (reused) {
           return;
         }
```

Don't download the model on re-use.